src/components/modal/Modal.jsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import './Modal.css';
+
+const Modal = ({
+  isOpen,
+  onClose,
+  confirmClose,
+  children
+}) => {
+  if (!isOpen) return null;
+
+  const handleClose = () => {
+    if (confirmClose) {
+      confirmClose().then((confirm) => {
+        if (confirm) onClose();
+      });
+    } else {
+      onClose();
+    }
+  };
+
+  const handleModalClick = (e) => {
```

Comment:
Do you really need this? It's not a good idea to stop the propagation of events. If I want to have a generic handler up on the component chain, you are breaking my expectations. 

Why are you doing this?